core/src/main/scala/kafka/coordinator/group/GroupMetadata.scala

```diff
@@ -167,10 +185,11 @@ case class CommitRecordMetadataAndOffset(appendedBatchOffset: Option[Long], offs
  *  3. leader id
  */
 @nonthreadsafe
-private[group] class GroupMetadata(val groupId: String, initialState: GroupState) extends Logging {
+private[group] class GroupMetadata(val groupId: String, initialState: GroupState, time: Time = Time.SYSTEM) extends Logging {
```

Comment:
We prefer not to have optional arguments (at least outside of test code) since they are too easy to overlook.

Reply:
Yes, I think it is better to require all arguments be explicitly provided.